### ISSUE TITLE
whereJsonSupersetOf does not take String as second argument

### ISSUE THREAD

**Issue by nkmnz** [knex/knex#4979]
# Environment

Knex version: 1.0.1
Database + version: pg 14.1 (latest docker image)
OS: macOS

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
According to the docs, whereJsonSupersetOf should take a column and, among others, a string to compare to. Passing a string results in an error when this string is not wrapped in brackets.

Doesn't work:
`await knex('table_name').whereJsonSupersetOf('path', "hello")`

these inputs work, though: 
`await knex('table_name').whereJsonSupersetOf('path', ["hello"])`
`await knex('table_name').whereJsonSupersetOf('path', 0)`
same results as: `await knex('table_name').whereJsonSupersetOf('path', [0])` or `await knex('table_name').whereJsonSupersetOf('path', '0')`

2. Error message

```
error: select * from "table_name" where "path" @> $1 - invalid input syntax for type json
    at Parser.parseErrorMessage (~/app/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (~/app/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (~/app/node_modules/pg-protocol/dist/parser.js:39:38) {
  length: 189,
  severity: 'ERROR',
  code: '22P02',
  detail: 'Token "hello" is invalid.',
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: "JSON data, line 1: hello\nunnamed portal parameter $1 = '...'",
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'jsonfuncs.c',
  line: '621',
  routine: 'json_ereport_error'
}
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

**Comment by OlivierCavadenti** [comment]
Seems to be this kind of issue : https://github.com/brianc/node-postgres/issues/442
I stringify the parameter, syntax error gone but the query returns nothing.
Still need investigating.

**Comment by mlevit** [comment]
Looks like I'm receiving the same error code for a simple update statement:

```
error: update "meta_migration" set "response" = $1 where "meta_migration_id" = $2 - invalid input syntax for integer: 
    "{"meta_migration_id":6}"
    Code: 22P02
```

**Comment by kibertoad** [comment]
Released in 1.0.2
